docs/rules/forbid-elements.md

```diff
@@ -0,0 +1,39 @@
+# Forbid certain elements (forbid-elements)
+
+You may want to forbid usage of certain elements in favor of others, (e.g. forbid all `<div />` and use `<Box />` instead). This rule allows you to configure a list of forbidden elements and to specify their desired replacements.
+
+## Rule Details
+
+This rule checks all JSX components and verifies that no forbidden elements are used. When it detects a forbidden element, the error message will contain the specified elements that you should use instead. This rule is off by default. If on, no elements are forbidden by default.
```

Comment:
I'm not sure we're on the same page here - `<Modal />` is identical to `React.createElement(Modal)`. `React.createElement('Modal')` is the same as `<{'Modal'} />` which is simply invalid react/jsx.

In other words, if `Button` is forbidden, i could still do `const NotButton = Button; <NotButton />` and work around this rule.